src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.Map.cs

```diff
@@ -193,6 +206,59 @@ public static void ReadMap_IncorrectDefiniteLength_ShouldThrowFormatException(st
             Assert.Throws<FormatException>(() => reader.ReadInt64());
         }
 
+        [Theory]
+        [InlineData("bf")]
+        [InlineData("bf0102")]
+        [InlineData("bf01020304")]
+        public static void ReadMap_IndefiniteLength_MissingBreakByte_ShouldReportEndOfData(string hexEncoding)
+        {
+            byte[] encoding = hexEncoding.HexToByteArray();
+            var reader = new CborReader(encoding);
+            reader.ReadStartMap();
+            while (reader.Peek() == CborReaderState.UnsignedInteger)
+            {
+                reader.ReadInt64();
+            }
+
+            Assert.Equal(CborReaderState.EndOfData, reader.Peek());
+        }
+
+        [Theory]
+        [InlineData("bf0102ff", 1)]
+        [InlineData("bf01020304ff", 2)]
+        [InlineData("bf010203040506ff", 3)]
+        public static void ReadMap_IndefiniteLength_PrematureEndArrayCall_ShouldThrowInvalidOperationException(string hexEncoding, int length)
+        {
+            byte[] encoding = hexEncoding.HexToByteArray();
+            var reader = new CborReader(encoding);
+            reader.ReadStartMap();
+            for (int i = 1; i < length; i++)
+            {
+                reader.ReadInt64();
+            }
+
+            Assert.Equal(CborReaderState.UnsignedInteger, reader.Peek());
+            Assert.Throws<InvalidOperationException>(() => reader.ReadEndMap());
+        }
+
+        [Theory]
+        [InlineData("bf01ff", 1)]
+        [InlineData("bf010203ff", 3)]
+        [InlineData("bf0102030405ff", 5)]
+        public static void ReadMap_IndefiniteLength_OddKeyValuePairs_ShouldThrowFormatException(string hexEncoding, int length)
+        {
+            byte[] encoding = hexEncoding.HexToByteArray();
+            var reader = new CborReader(encoding);
+            reader.ReadStartMap();
+            for (int i = 0; i < length; i++)
+            {
+                reader.ReadInt64();
+            }
+
+            Assert.Equal(CborReaderState.EndMap, reader.Peek()); // don't want this to fail
+            Assert.Throws<FormatException>(() => reader.ReadEndMap());
```

Comment:
This feels weird to me.  But the best answer I can come up with is adding a new state of "IncompleteMap".

Reply:
I suppose you mean the first assertion, right?

Reply:
The pair.  "EndMap" feels like "ReadEndMap will succeed"; but the test says otherwise.

Reply:
Perhaps we could use a nondescript `FormatError` instead?

Reply:
`FormatError` / `InvalidData`, something of that nature seems reasonable.

Reply:
Done